File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> count = new HashMap<>();
        for(int num:nums) count.put(num, count.getOrDefault(num, 0)+1);
        PriorityQueue<int[]> pq = new PriorityQueue<>((a,b)-> a[1]-b[1]);
        for(int key:count.keySet()){
            pq.add(new int[]{key, count.get(key)});
            if(pq.size()>k){
                pq.poll();
            }
        }
        int res[] = new int[k];
        int p = 0;
        while(!pq.isEmpty()) res[p++] = pq.poll()[0];
        return res;
    }
}

File: Data Structures & Algorithms/kth-largest-integer-in-a-stream/submission-2.cs
public class KthLargest {
    PriorityQueue<int, int> pq;
    int k;
    public KthLargest(int k, int[] nums) {
        this.k = k;
        this.pq = new PriorityQueue<int, int>();
        foreach(int num in nums){
            Add(num);
        }
    }
    
    public int Add(int val) {
        this.pq.Enqueue(val,val);
        if(pq.Count>k) pq.Dequeue();
        return pq.Peek();
    }
}
